openeogeotrellis/vectorcube.py

```diff
@@ -39,49 +40,61 @@ def run_udf(self, udf: str, *, runtime: str = "Python", context: Optional[dict]
         # TODO: leverage `runtime` argument?
         udf_globals = openeo.udf.run_code.load_module_from_string(code=udf)
 
+
+
+        csv_paths = list(Path(self._csv_dir).glob("*.csv"))
+        _log.info(
+            f"{type(self).__name__} run_udf with {self._csv_dir=} ({len(csv_paths)=})"
+        )
+
+
+        spark = SparkSession.builder.master('local[1]').getOrCreate()
+        df = spark.read.csv([f"file://{p}" for p in csv_paths],header=True)
+
+
+        columns = df.columns
+
         # TODO: Port this UDF detection to openeo.udf.run_code?
         if "udf_apply_udf_data" in udf_globals:
             udf_function = udf_globals["udf_apply_udf_data"]
             callback = self._get_run_udf_udf_data_callback(
-                udf_function=udf_function, context=context
+                udf_function=udf_function, context=context,columns = columns
             )
         elif "udf_apply_feature_dataframe" in udf_globals:
             udf_function = udf_globals["udf_apply_feature_dataframe"]
             callback = self._get_run_udf_pandas_callback(
-                udf_function=udf_function, context=context
+                udf_function=udf_function, context=context, columns=columns
             )
         else:
             raise openeo.udf.OpenEoUdfException("No UDF found")
 
-        csv_paths = list(Path(self._csv_dir).glob("*.csv"))
-        _log.info(
-            f"{type(self).__name__} run_udf with {self._csv_dir=} ({len(csv_paths)=})"
-        )
-        import pyspark.pandas
-        #pyspark.pandas.set_option('compute.default_index_type', 'distributed')
-        #pyspark.pandas.set_option('compute.shortcut_limit', 1)
-        #pyspark.pandas.set_option('compute.max_rows', None)
+        id_index = columns.index("feature_index")
+
+        def mapTimeseriesRows(id_bands):
+
 
-        csv_df = pyspark.pandas.read_csv([f"file://{p}" for p in csv_paths])
+            result = callback(id_bands)
+            if isinstance(result,pd.Series):
+                result = pd.DataFrame(result).T
+            else:
+                result.reset_index(inplace=True)
+            result["feature_index"] = pd.to_numeric(id_bands[0])
+            result.insert(0, 'feature_index', result.pop('feature_index'))
+            return result.to_dict(orient='tight')
 
-        processed_df = csv_df.groupby("feature_index").apply(callback).reset_index()
+        csv_as_list = df.rdd.map(list).map(lambda x: (x[id_index],x)).groupByKey().map(mapTimeseriesRows)
```

Comment:
```suggestion
        csv_as_list = df.rdd.map(list).map(lambda x: (x[id_index],x)).groupByKey().map(map_timeseries_rows)
```

Reply:
I'm still figuring this out, but `csv_as_list` seems bit confusing, this isn't a list of csv anymore, but an RDD of dicts, right?